ejb3/src/main/java/org/jboss/as/ejb3/cache/simple/SimpleCache.java

```diff
@@ -169,6 +169,8 @@ public void release(V bean) {
                     // However, removing it directly is faster than scheduling it for immediate removal.
                     remove(id);
                 }
+            } else {
+                remove(id);
```

Comment:
This doesn't look right.  This would mean that a @Stateful EJB never retains its state across multiple invocations if it doesn't define a @StatefulTimeout.

Reply:
I'll see, I'll skip this one